brave/brave-core
# (Wallet) Implement fingerprint unlock screen

## Resolves https://github.com/brave/brave-browser/issues/38987

Implements new fingerprint setup screen that will be shown during Wallet onboarding process.

The new UI follows provided [Figma design](https://www.figma.com/design/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?node-id=6223-70942&m=dev).

### UI Preview

|  Light Theme  | Dark Theme  |
|---|---|
| <img src="https://github.com/brave/brave-core/assets/3308503/7aa84331-43fd-428b-a86b-5ba6e65e49c6" width="300" />  | <img src="https://github.com/brave/brave-core/assets/3308503/47c8085f-9dbb-4f3a-802c-937ef5cc1c89" width="300" />  |



### Nala Tokens

Styles and attributes introduced have been copied from exported Nala tokens:

- Color `container_background`
- Style `HeadingH3`
- Style `DefaultRegular`
- Color `text_primary`
- Color `text_secondary`
- Dimension `heading_h3`

### New Strings

The new strings have been modified and adapted to follow proper Android naming convention as the term "Touch ID" presented in the Figma design is iOS specific and not present in the Android ecosystem.

The right term to use following Google guidelines is "biometric authentication": https://developer.android.com/identity/sign-in/biometric-auth

The name of the technology that would allow your fingerprint to unlock the smartphone is called "Fingerprint Unlock": https://guidebooks.google.com/pixel/optimize-your-life/set-up-a-fingerprint

### Enhancements

The previous implementation had a few issues resolved by this PR:
- A toast notification with an empty string was constantly popping up when finishing the setup.
- Cancelling the fingerprint setup was erroneously proceeding to the next screen without any option to retry.




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Clear Wallet data, if any
- Navigate to Wallet section to trigger onboarding flow
- Reach fingerprint setup scree
- Tap on "Use Fingerprint Unlock" 
- Observe system dialog appears asking to confirm the fingerprint
- Tap on cancel button
- Observe the dialog disappear without navigating to the new screen
- Tap on again on "Use Fingerprint Unlock"
- Touch and hold the finger on the sensor
- Observe the Wallet is created correctly
